packages/ember-utils/lib/to-string.js

```diff
@@ -1,11 +1,33 @@
 const objectToString = Object.prototype.toString;
 
+function isNone(obj) {
+  return obj === null || obj === undefined;
+}
+
 /*
  A `toString` util function that supports objects without a `toString`
  method, e.g. an object created with `Object.create(null)`.
 */
 export default function toString(obj) {
-  if (obj && typeof obj.toString === 'function') {
+  if (Array.isArray(obj)) {
+    // Reimplement Array.prototype.join according to spec (22.1.3.13)
+    // Changing ToString(element) with this safe version of ToString.
+    let len = obj.length;
+    let sep = ',';
+    let r = '';
+
+    for (let k = 0; k < len; k++) {
+      if (k > 0) {
+        r += ',';
+      }
+
+      if (!isNone(obj[k])) {
+        r += toString(obj[k]);
+      }
+    }
+
+    return r;
+  } else if (obj && typeof obj.toString === 'function') {
```

Comment:
`typeof obj === 'object' && !== obj && typeof obj.toString === 'function'` should do the trick